File: leetcode/172-factorial-trailing-zeroes/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int trailingZeroes(int n) {
        int count = 0;
        while (n >= 5) {
            n /= 5;
            count += n;
        }
        return count;
    }
};

int main() {
    Solution sol;
    cout << sol.trailingZeroes(5) << endl;  // 1
    cout << sol.trailingZeroes(10) << endl; // 2
    cout << sol.trailingZeroes(25) << endl; // 6
    return 0;
}

File: leetcode/172-factorial-trailing-zeroes/solution.py
class Solution:
    def trailingZeroes(self, n: int) -> int:
        count = 0
        while n >= 5:
            n //= 5
            count += n
        return count


if __name__ == "__main__":
    print(Solution().trailingZeroes(5))   # 1
    print(Solution().trailingZeroes(10))  # 2
    print(Solution().trailingZeroes(25))  # 6
